BUG: fix f2py shebang line for bdist wheel, egg

matthew-brett opened the pull request:
This is a painful hack, but I don't know a better way.  So, I put it forward
for improvement by y'all.

Command `bdist_wheel` was generating a shebang line for f2py that uses the
Python path for the building Python.  If we are building a wheel or an egg,
use the generic `#!python` shebang line for the f2py script instead, which
setuptools will modify at install time.

Closes gh-5812.

matthew-brett committed:
BUG: fix f2py shebang line for bdist wheel, egg

Command `bdist_wheel` was generating a shebang line for f2py that uses
the Python path for the building Python.  If we are building a wheel or
an egg, use the generic `#!python` shebang line for the f2py script
instead, which setuptools will modify at install time.

Closes gh-5812.

rgommers commented on numpy/f2py/setup.py:
```diff
@@ -29,6 +29,20 @@
+    should be ``#!python`` rather than ``#!`` followed by the contents of
+    ``sys.executable``.
+    """
+    import IPython; IPython.embed()
```
left-over line from debugging here

matthew-brett replied:
Oh dear - thanks for catching that.

rgommers commented:
It doesn't look that hacky to me. ``#!python`` should work. I think that the Windows installers have the same issue (haven't checked, but has to be the case). We create those with ``bdist_wininst``, so would be good to add that string as well. And while we're at it, ``bdist_mpkg`` for the OS X dmg installers.

matthew-brett committed:
ENH: add bdist_mpkg, bdist_wininst to binary dists

Add other binary distribution formats to list of build commands that
should generate !python shebang lines.

matthew-brett commented:
It would be good to test this is working - here's a first pass : https://github.com/numpy/numpy/pull/5817

matthew-brett commented:
https://github.com/numpy/numpy/pull/5817 now working I believe - so if you merge this PR, it might be worth considering #5817 to test this fix.

charris commented:
Merged, thanks Matthew.
